src/test/integration/addons-test.js

```diff
@@ -9,22 +9,62 @@ const {
   createUser,
   headerAuth,
 } = require('../user');
-const Constants = require('../../constants');
+
 const AddonManager = require('../../addon-manager');
-const pFinal = require('../promise-final');
-const fetch = require('node-fetch');
 const config = require('config');
+const Constants = require('../../constants');
+const fetch = require('node-fetch');
+const jsonfile = require('jsonfile');
+const path = require('path');
+const pFinal = require('../promise-final');
+
+const testManifestFilename = path.join(__dirname, '../..',
+                                       config.get('addonManager.path'),
+                                       'test-adapter', 'package.json');
+
+const testManifest = {
+  name: 'test-adapter',
+  moziot: {
+    api: {
+      min: 1,
+      max: 1
+    },
+    enabled: true,
+  },
+};
+
+function copyManifest(manifest) {
+  // This essentially does a deep copy.
+  return JSON.parse(JSON.stringify(manifest));
```

Comment:
This would be more efficient:
`return Object.assign({}, manifest);`

Reply:
Unfortunately, that only does a shallow copy and modifications made to nested objects in the returned manifest cause changes to testManifest. Using the JSON trick is a cheap deep copy. Object.assign is NOT a deep copy.

Reply:
Ah, good point.